compiler/optimizer/SwitchAnalyzer.cpp

```diff
@@ -1343,7 +1343,7 @@ int32_t *TR::SwitchAnalyzer::setupFrequencies(TR::Node *node)
       TR::Node *caseNode = node->getChild(i);
       TR::Block *targetBlock = caseNode->getBranchDestination()->getNode()->getBlock();
       int32_t targetCount = targetCounts[targetBlock->getNumber()];
-      TR_ASSERT(targetCount != 0, "unreachle successor of switch statement");
+      TR_ASSERT_FATAL(targetCount > 0, "Successor block_%d of switch statement has non-sense successsor count", targetBlock->getNumber());
```

Comment:
Thinking about this a little, would it make sense to make `targetCount` (and `targetCounts`) unsigned?

Also, as a general word of caution: the C++ compiler is allowed to "optimize" the assert condition into `targetCount != 0` because signed overflow is undefined behaviour. In theory, if it figured out that target counts start at zero and are only incremented, the compiler is allowed to assume signed overflow never happens.